### DISCUSSION TITLE
How do I handle overloaded functions?

### DISCUSSION THREAD

**Discussion by LiliumAtratum** [michaeljones/breathe#903] [Q&A]
I am looking for an answer, but I couldn't find one in the documentation.
I have two functions like this:

```
template<typename A, typename... Args>
A make(Args... args)

template<typename... Args>
auto make(Args... args)
```

I would like to include both of those functions in my documentation with the least amount of effort. Currently I just have:

```
.. doxygenfunction:: make
```
but this gives me a warning, that it doesn't know which version of the function I want to include (I want both, in any order).

When looking in the documentation for this, all I could find is this weird "Test Pages / Doxygen Test Suite" page that kind-of shows that it may be possible, but does not explain how to achieve it. I tried looking in the sources for this particular section, but it left me even more confused. All I found this this:

```
.. cpp:namespace:: @ex_doxygen_overload
.. doxygenindex::
   :path: ../../examples/doxygen/overload/xml
```

So, how do I use it?

**Comment by vermeeren** [comment]
@LiliumAtratum I believe you can specify the full type to the `doxygenfunction` directive. iirc at some point in time the warning was changed to include the options Breathe/Sphinx sees, could you post the full warning/error given?

From memory, the exact syntax reported as potential candidates should work, but it's been a while since I last checked this.

**Reply by LiliumAtratum** [reply]
I wish I didn't have to specify full type, if I actually want _all_ versions.
Still, even if I try, I fail to.
The reported error is as follows:
```
operations.rst:4: WARNING: doxygenfunction: Unable to resolve function "make" with arguments None in doxygen xml output [...]
Potential matches:
- template<typename ...Args> auto make(Args... args)
- template<typename Vector, typename ...Args> Vector make(Args... args)
```

Unfortunately, copy-pasting the reported matches fails to work. For example:

```
WARNING: doxygenfunction: Cannot find function "template<typename... Args> auto make" in doxygen xml output [...]
```

**Comment by LiliumAtratum** [comment]
Almost a year later, still hoping to find a solution to this...
